colcon_core/shell/template/prefix.bat.em

```diff
@@ -70,6 +70,23 @@ goto:eof
   if "%COLCON_PYTHON_EXECUTABLE%" NEQ "" (
     set "_colcon_python_executable=%COLCON_PYTHON_EXECUTABLE%"
   )
+  if not exist "%_colcon_python_executable%" (
+    rem provided colcon_python_executable not found
+    rem falling back to build time python
+    if not exist "%(python_executable)" (
```

Comment:
I would suggest not to check for the compile time `python_executable` again. That was already the starting value when not overridden with an environment variable. I would keep the logic as close as possible to what is done in `sh`: https://github.com/colcon/colcon-core/blob/aca10c55a6c7749777171bb18fce20ea27ca0d1d/colcon_core/shell/template/prefix.sh.em#L67-L82

Reply:
This means that if the user provides an invalid override we will never fall back to the python used at compile time.

Though it seems more likely that the one used for compiling will have more change to provide expected results than the one on the path that may be a different version (that could actually be python 2 on windows with the current patch).

Can you clarify why we should not fall back to the compile time one if available?

Reply:
> Can you clarify why we should not fall back to the compile time one if available?

If the user provide a Python path it probably shouldn't even try any fall back. Because the user asked explicitly to use that. The `sh` logic could be changed to not try a fall back in that case.